freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -181,4 +192,30 @@ private boolean isDuplicateEntryException(FreeIpaClientException e) {
                 .filter(c -> c == DUPLICATE_ENTRY_ERROR_CODE)
                 .isPresent();
     }
+
+    SecretResponse getSecretResponseForPrincipal(String accountId, String principal) {
+        try {
+            String path = String.format("%s/%s/%s/%s-%s", accountId,
+                    "ServiceKeytab", "serviceprincipal",
+                    UUID.randomUUID().toString(), Long.toHexString(System.currentTimeMillis()));
+            String secret = secretService.put(path, principal);
+            return stringToSecretResponseConverter.convert(secret);
+        } catch (Exception exception) {
+            LOGGER.warn("Something went wrong with Secret store. ServiceKeytabResponse is not updated ", exception);
+        }
+        return null;
```

Comment:
The other methods have been modified to throw `KeytabCreationException` rather than returning `null`. It seems like returning `null` might ignore the swallowed exceptions.

(Same for `getSecretResponseForKeytab()`)